app/src/main/java/tech/relaycorp/courier/data/preference/StoragePreferences.kt

```diff
@@ -0,0 +1,27 @@
+package tech.relaycorp.courier.data.preference
+
+import androidx.annotation.VisibleForTesting
+import com.tfcporciuncula.flow.FlowSharedPreferences
+import kotlinx.coroutines.flow.asFlow
+import javax.inject.Inject
+import javax.inject.Provider
+
+class StoragePreferences
+@Inject constructor(
+    private val preferences: Provider<FlowSharedPreferences>
+) {
+
+    private val maxStorage by lazy {
+        preferences.get().getInt("max_storage", DEFAULT_MAX_STORAGE_PERCENTAGE)
```

Comment:
I'm not sure whether we should be storing a percentage or the number of MiB/GiB, but I think we should store whatever unit the user sees in the preferences screen. I think the distinction matters in devices that support external (micro)SD cards.

Either way, I think the preference (`max_storage`) should include the unit (e.g., `max_storage_percentage`) to avoid ambiguity.

Reply:
Good question, if the user picks a percentage (but we show the total), or it picks an absolute number (limited by the total device storage). Either way, I was thinking of showing a dropdown in the settings, instead of letting the user enter a possibly invalid value.

You are right about the units, but I'm thinking of using types instead of names to make ir clear.